# STAT 207 Lab 7: Hypothesis Testing - [25.5 total points] (0.5) is bonus points - Solutions

Due: Wednesday, March 24 11:59pm CST



## <u>Case Study</u>: U.S Unemployment Rates and Median Education Levels
In this lab we will explore the statistical concepts that we've learned this week while also examining the **unemployment rates** and **median education levels** of the counties in the U.S. 

### Imports

In [1]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import numpy as np

## 1. <u>Preliminaries</u>

### Read the county.csv into a dataframe, call it df, and display the first five rows.

Remember that 'data unavailable' represents missing values in this csv. Make sure you encode these values as NaN when you read the data in.

In [2]:
df = pd.read_csv('county.csv')
df.head()

name    state pop2000  pop2010 pop2017 pop_change poverty  \
0  Autauga County  Alabama   43671    54571   55504       1.48    13.7   
1  Baldwin County  Alabama  140415   182265  212628       9.19    11.8   
2  Barbour County  Alabama   29038    27457   25270      -6.22    27.2   
3     Bibb County  Alabama   20826    22915   22668       0.73    15.2   
4   Blount County  Alabama   51024    57322   58013       0.68    15.6   

   homeownership  multi_unit unemployment_rate metro    median_edu  \
0           77.5         7.2              3.86   yes  some_college   
1           76.7        22.6              3.99   yes  some_college   
2           68.0        11.1               5.9    no    hs_diploma   
3           82.9         6.6              4.39   yes    hs_diploma   
4           82.0         3.7              4.02   yes    hs_diploma   

  per_capita_income median_hh_income smoking_ban  
0           27841.7            55317        none  
1          27779.85            52562        none  
2          17891.73            33368     partial  
3          20572.05            43404        none  
4          21367.39            47412        none

In [3]:
df.dtypes

name                  object
state                 object
pop2000               object
pop2010                int64
pop2017               object
pop_change            object
poverty               object
homeownership        float64
multi_unit           float64
unemployment_rate     object
metro                 object
median_edu            object
per_capita_income     object
median_hh_income      object
smoking_ban           object
dtype: object

In [4]:
for i in df['pop2000'].unique():
    print(i)
#found non numbers: data unavailable

43671
140415
29038
20826
51024
11714
21399
112249
36583
23988
39593
15922
27867
14254
14123
43615
54984
14089
12202
37631
13665
77483
49129
46365
64452
65874
38440
103459
18495
31223
25764
9974
17185
16310
88787
53926
662047
15904
87966
34803
115092
65676
13473
24105
276700
22539
31214
82231
399843
24324
223510
111064
11861
20949
29605
22380
49756
64742
143293
14798
80321
41475
164875
70713
18097
13183
24843
2697
5465
260283
16006
1258
1893
4922
82840
2392
3436
30711
49691
14070
13913
1823
59322
9196
7385
7208
6684
6146
8835
data unavailable
6174
10195
7028
808
6551
69423
117755
116320
51335
33489
8547
19715
3072149
155032
97470
843746
179727
38381
167517
160026
20749
24209
38386
153406
33948
12600
5744
25357
14117
23546
17609
24046
8571
25603
20336
82148
53247
50866
19526
9210
15341
18723
86014
17771
11642
88068
16464
37331
23587
30353
14300
34233
13249
18418
84278
22781
8559
17774
12580
14492
13628
22486
52828
14243
16140
40443
51979
10254
9245
9955
8608
28790
10209
26445
11303
25614

55716
2158
18837
1977
6851
1932
3884
95802
4497
15694
493
4601
6424
1858
7180
1199
36070
9667
10620
9383
10227
4105
34606
8195
3609
5267
861
7675
2259
1068
129352
31151
7452
444
819
583
6259
12158
2438
3525
42259
8767
24334
9615
4068
6148
9830
7039
10441
10203
11793
20253
9060
24365
2098
6339
36160
463585
2292
6634
3574
3099
5324
22993
1902
2143
747
2714
53534
9403
3786
3111
11551
783
6567
8333
4488
6882
8875
983
4089
9374
250291
34632
774
712
533
35226
8204
5440
4038
7576
5057
15396
3087
3200
9747
7857
31662
5639
11448
9531
1756
13843
122595
19830
36951
16496
6198
3318
1475
6455
6055
729
7171
4647
18780
9851
4061
886
14598
23982
1375765
41259
45291
971
1651
16106
5794
4165
34501
5071
32485
6693
3399
339486
9181
52457
56325
43666
73825
33111
81743
380841
136225
277359
112233
40458
252552
884118
423394
508932
102326
146438
793633
254673
608975
121989
350761
750162
615301
470212
510916
489049
64285
297490
144166
522541
102437
556678
3543
61382
14189
45044
2240
174682
51658
31002
4680
810

In [5]:
missing_values = ['data unavailable']
df = pd.read_csv('county.csv', na_values = missing_values)

In [6]:
df.dtypes

name                  object
state                 object
pop2000              float64
pop2010                int64
pop2017              float64
pop_change           float64
poverty              float64
homeownership        float64
multi_unit           float64
unemployment_rate    float64
metro                 object
median_edu            object
per_capita_income    float64
median_hh_income     float64
smoking_ban           object
dtype: object

In [7]:
df.head()

name    state   pop2000  pop2010   pop2017  pop_change  poverty  \
0  Autauga County  Alabama   43671.0    54571   55504.0        1.48     13.7   
1  Baldwin County  Alabama  140415.0   182265  212628.0        9.19     11.8   
2  Barbour County  Alabama   29038.0    27457   25270.0       -6.22     27.2   
3     Bibb County  Alabama   20826.0    22915   22668.0        0.73     15.2   
4   Blount County  Alabama   51024.0    57322   58013.0        0.68     15.6   

   homeownership  multi_unit  unemployment_rate metro    median_edu  \
0           77.5         7.2               3.86   yes  some_college   
1           76.7        22.6               3.99   yes  some_college   
2           68.0        11.1               5.90    no    hs_diploma   
3           82.9         6.6               4.39   yes    hs_diploma   
4           82.0         3.7               4.02   yes    hs_diploma   

   per_capita_income  median_hh_income smoking_ban  
0           27841.70           55317.0        none  
1           27779.85           52562.0        none  
2           17891.73           33368.0     partial  
3           20572.05           43404.0        none  
4           21367.39           47412.0        none

### Create a series that is comprised of just the unemployment rate variable and drop any missing values from this dataset.

In [8]:
unem = pd.DataFrame(df['unemployment_rate'])
unem

#is dataframe here okay or does it need to be a series? does it make a difference?

unemployment_rate
0                  3.86
1                  3.99
2                  5.90
3                  4.39
4                  4.02
...                 ...
3137               4.55
3138               2.99
3139               4.50
3140               4.08
3141               3.98

[3142 rows x 1 columns]

In [9]:
unem.isna().sum()

unemployment_rate    3
dtype: int64

In [10]:
unem.dropna(inplace = True)
unem.isna().sum()

unemployment_rate    0
dtype: int64

In [11]:
n_unem = unem.shape[0]
n_unem

3139

### Create a series that is comprised of just the median education variable and drop any missing values from this dataset.

In [12]:
med_edu = pd.DataFrame(df['median_edu'])
med_edu

median_edu
0     some_college
1     some_college
2       hs_diploma
3       hs_diploma
4       hs_diploma
...            ...
3137  some_college
3138     bachelors
3139  some_college
3140  some_college
3141  some_college

[3142 rows x 1 columns]

In [13]:
med_edu.isna().sum()

median_edu    2
dtype: int64

In [14]:
med_edu.dropna(inplace = True)
med_edu.isna().sum()

median_edu    0
dtype: int64

In [15]:
n_med_edu = med_edu.shape[0]
n_med_edu

3140

## 2. <u>Unemployment Rate Population Analysis</u>

We will assume that the series of all county unemployment rates from 1b is our **population**. Usually, we do not have the whole population at our disposal, which is why we conduct inference in the first place.

However, when conducting our hypothesis testing in this assignment we will assume that we don't have this population information. After we have conducted the hypothesis testing, we can use our population information to "check" our answers to learn more about how hypothesis testing works.

### 2a. [0.5 pt] Calculate the mean and standard deviation unemployment rate of ALL U.S. counties.

In [16]:
unem_mean = unem['unemployment_rate'].mean()
unem_mean

4.611443134756292

In [17]:
unem_std = unem['unemployment_rate'].std()
unem_std

1.648761085622538

### 2b.  [1 pt] Calculate the z-score of the unemployment rate of 'Champaign County' in the state of 'Illinois'.

In [18]:
champaign = df[df['name'] == 'Champaign County']
#returns dataframe with two champaign county values -- value for illinois is 4.24
champaign

name     state   pop2000  pop2010   pop2017  pop_change  \
604   Champaign County  Illinois  179669.0   201081  209399.0        1.83   
2053  Champaign County      Ohio   38890.0    40097   38840.0       -1.54   

      poverty  homeownership  multi_unit  unemployment_rate metro  \
604      21.4           55.9        35.2               4.24   yes   
2053     11.1           75.3        14.7               4.16    no   

        median_edu  per_capita_income  median_hh_income smoking_ban  
604   some_college           29619.49           49586.0         NaN  
2053    hs_diploma           25704.80           54495.0         NaN

In [19]:
champaign = 4.24
champaign

4.24

In [20]:
z_score_champ = (champaign - unem_mean)/unem_std
print('Test Statistic: ', z_score_champ)

Test Statistic:  -0.2252862091392959


## 3. <u>Unemployment Rate Sample Analysis</u>

### 3a.  [0.5 pt] Collect a random sample of size 35 from your population of all county unemployment rates (without replacement) using a random state of 3.

In [21]:
sample = unem.sample(35, replace = False, random_state = 3)
sample

unemployment_rate
340                5.18
983                3.80
2604               3.85
2975               6.59
2116               4.77
2419               3.33
1127               7.88
2279               3.82
30                 4.27
564                3.55
288                2.44
472                5.11
372                4.65
1667               2.27
2153               3.26
1490               5.18
2500               4.36
3118               3.76
2197               5.86
553                3.01
802                2.31
2651               3.52
770                3.36
634                4.82
942                2.14
1338               3.25
3038               5.12
2107               4.59
3006               6.09
1173               4.63
619                3.75
969                3.83
2350               6.56
2178               4.70
894                3.21

### 3b.  [0.5 pt] Calculate the mean and standard deviation of this sample.

In [22]:
sample_mean = sample['unemployment_rate'].mean()
sample_mean

4.252

In [23]:
sample_std = sample['unemployment_rate'].std()
sample_std

1.3095751443974222

## 4. <u>Unemployment Rate Sampling Distribution Analysis</u>

Assume you know $\sigma$ in this part.

### 4a.  [0.5 pt] What is the mean and standard deviation of the sampling distribution of sample mean unemployment rates (collected from samples of size 35)?

the mean would be $\mu\$

In [24]:
unem_mean

4.611443134756292

the standard deviation would be $\sigma\$/sqrt(n)

In [25]:
unem_std/np.sqrt(35) #make sure to use sample size not population size

0.2786914893089392

### 4b.  [1 pt] Is the sampling distribution of samples means (collected from samples of size 35) approximately normal? Explain why or why not?

Yes, the sample was collected randomly, n < 10% of the population size, and n is larger than 30

### 4c.  [1 pt] What is the z-score of our sample mean from 3b?

In [26]:
assumed_pop_mean = unem_mean
n_unem1 = 35



print('Assumed Population Mean: ', assumed_pop_mean)

std_error1 = unem_std/np.sqrt(n_unem1)
print("Standard Error: ", std_error1)


Assumed Population Mean:  4.611443134756292
Standard Error:  0.2786914893089392


In [27]:
z_score_4c = (sample_mean - assumed_pop_mean)/std_error1
print('Z-score: ', z_score_4c)

Z-score:  -1.2897528218302958


### 4d.  [1 pt] What is the probability that a randomly selected sample mean z-score is greater than or equal to the z-score that we calculated in 4c?

In [28]:
from scipy.stats import norm

1 - norm.cdf(z_score_4c, loc = 0, scale = 1)

0.9014317534230412

### 4e.  [0.5 pt] What is the probability that a randomly selected sample mean z-score is less than or equal to the negative absolute value of the z-score that we calculated in 4c?

In [29]:
norm.cdf(-np.abs(z_score_4c), loc = 0, scale = 1)

0.0985682465769589

## 5. <u>Hypothesis Testing for the Average Unemployment Rate of ALL US Counties</u>

For now, suppose that we know nothing about our population that we explored in #2.


Suppose that a reliable source suggested that the average unemployment rate of all US counties is 4.611. We are skeptical of this source, and would like to test this claim and see if there is evidence to suggest that this is not the case. We will use our random sample that we collected from the population in #3 to test this claim.

### 5a.  [1 pt] Formulate your null and alternative hypothses for this hypothesis test.

$h_0$ : $\mu$ = 4.611



$h_a$ : $\mu$ $\neq$ 4.611

### 5b.  [1 pt] Check the conditions for conducting hypothesis testing using the sample from #3.



The sample was collected randomly, n < 10% of the population size, and n is larger than 30

### 5c.  [1 pt] Calculate the p-value for this test.

Assume you do not know $\sigma$.

In [30]:
assumed_pop_mean1 = 4.611
n_unem1 = 35

In [31]:
std_error2 = sample_std/np.sqrt(n_unem1)
print('Standard Error: ', std_error2)

test_statistic2 = (sample_mean - assumed_pop_mean)/std_error2
print('Test Statistic: ', test_statistic2)

Standard Error:  0.22135860103483856
Test Statistic:  -1.6238046910123047


In [32]:
p_value = 2 * (1 - norm.cdf(np.abs(test_statistic2), loc = 0, scale = 1))
print('P-value: ', p_value)

P-value:  0.10441749858303595


### 5d.  [1 pt] Make a conclusion about your hypotheses using this p-value and a significance level of $\alpha=0.05$.

Becuase p-value = 0.10441749858303595 is > $\alpha\$ = 0.05, we fail to reject the null hypothesis. There is insufficient evidence to suggest the alternative hypothesis.

### 5e.  [1 pt] Now we would like to make a 95% confidence interval for the average unemployment rate of al U.S. counties. First, check the conditions for creating this confidence interval.

Sample is collected randomly, n < 10% of population size, and n > 30

### 5f.  [1 pt] Now create this 95% confidence interval.

Assume you do not know $\sigma$.

In [33]:
alpha = 0.05
critical_value = norm.ppf(1 - (alpha/2))
#equivalent to norm.ppf(.975)

print('Critical Value: ', critical_value)


ll = sample_mean - critical_value * (sample_std/np.sqrt(n_unem1))
ul = sample_mean + critical_value * (sample_std/np.sqrt(n_unem1))
    #n_unem1 = 35
    #sample_mean = 4.252


print('95% confidence interval:', '(',ll,',',ul,')')



Critical Value:  1.959963984540054
95% confidence interval: ( 3.8181451143035456 , 4.685854885696454 )


### 5g.  [1 pt] Use this confidence interval to make a conclusion about your hypothesis tests.

Becuase $\mu\$ = 4.611 is inside the confidence interval, we fail to reject the null hypothesis. There is insufficient evidence to suggest the alternative hypothesis

### 5h.  [0.5 pt] NOW, using what we *actually* know about the population and population mean, did our hypothesis test *actually* make the right conclusions in 5d and 5g?

5d and 5g both fail to reject the null , so our hypothesis test did make the right conclusion becuase they both agree

## 6. <u>Median Education Level Population Analysis</u>

We will assume that the series of all county median education levels from 1c is our **population**. Usually, we do not have the whole population at our disposal, which is why we conduct inference in the first place.

However, when conducting our hypothesis testing in this assignment we will assume that we don't have this population information. After we have conducted the hypothesis testing, we can use our population information to "check" our answers to learn more about how hypothesis testing works.



### <u>Tutorial</u>: Conditional Filtering of a Series

In [34]:
import pandas as pd
import numpy as np
small_series=pd.Series(['stat107', 'stat207', 'stat307', 'stat207'])
small_series

0    stat107
1    stat207
2    stat307
3    stat207
dtype: object

We can count the number of entries in a series that meet a certain condition by setting up a condition upon the values in a series and taking the sum of the Boolean values.

In [35]:
small_series=='stat207'

0    False
1     True
2    False
3     True
dtype: bool

In [36]:
np.sum(small_series=='stat207')

2

### 6a.  [0.5 pt] Calculate the proportion of all U.S. counties that have a median education level of 'hs_diploma'.

In [37]:
prop_hs = np.sum(med_edu['median_edu'] == 'hs_diploma')
prop_hs

1397

In [38]:
prop_hs = prop_hs/med_edu.shape[0]
prop_hs

0.4449044585987261

## 7. <u>Median Population Sample Analysis</u>

### 7a.  [0.5 pt] Collect a random sample of 40 county median education levels from the population of ALL U.S. county median education levels, (without replacement), using a random state of 100.

In [39]:
med_edu_sample = med_edu.sample(40, replace = False, random_state = 100)
med_edu_sample

median_edu
1168    hs_diploma
1049  some_college
1921  some_college
1740  some_college
3138     bachelors
497   some_college
1745  some_college
1148  some_college
1248    hs_diploma
1333  some_college
1907  some_college
1738  some_college
117     hs_diploma
2711    hs_diploma
617   some_college
781     hs_diploma
1276  some_college
952     hs_diploma
2444    hs_diploma
2746  some_college
1669  some_college
1079    hs_diploma
2022  some_college
2342    hs_diploma
154     hs_diploma
3004    hs_diploma
2140    hs_diploma
29      hs_diploma
31      hs_diploma
322   some_college
2027  some_college
1638  some_college
357     hs_diploma
1989    hs_diploma
775   some_college
2089  some_college
3036    hs_diploma
2825    hs_diploma
1162    hs_diploma
1137    hs_diploma

### 7b.  [0.5 pt] Calculate the proportion of counties in this sample that have a median education level of high school diploma.

In [40]:
prop_sample_hs = np.sum(med_edu_sample['median_edu'] == 'hs_diploma')
prop_sample_hs = prop_sample_hs/med_edu_sample.shape[0]
prop_sample_hs

0.5

## 8. <u>Sampling Distribution of Sample Proportions Analysis</u>



### 8a. [1 pt] What is the mean and standard deviation of the sampling distribution of sample propotion of counties with a median education level of high school diploma (collected from samples of size 40)?

Assume you actually know the population proprtion $p$ in part 8.

the population mean is \$p\$

In [41]:
prop_hs

0.4449044585987261

the standard deviation is sqrt((\$p\$ (1 - \$p\$))/n)

In [42]:
num = prop_hs * (1-prop_hs)
den = med_edu.shape[0]

np.sqrt(num/den)

0.008868546030771033

### 8b.  [1 pt] What is the z-score of our sample proportion from 7b?

In [43]:
n_prop = 40

assumed_pop_prop = prop_hs
print('Assumed Population Proportion:', assumed_pop_prop)

std_error_prop = np.sqrt((assumed_pop_prop * (1 - assumed_pop_prop))/n_prop)
print('Standard Error: ', std_error_prop)



Assumed Population Proportion: 0.4449044585987261
Standard Error:  0.07857551802528899


In [44]:
z_score8b = (prop_sample_hs - assumed_pop_prop)/std_error_prop
print('z-score: ', z_score8b)

z-score:  0.7011794867650984


## 9. <u>Hypothesis Testing for the Proportion of ALL U.S. Counties that have a Median Education Level of High School Diploma</u>

For now, suppose that we know nothing about our population that we explored in #6.

Previous studies have suggested that in the past, 45% of U.S. counties have a median education level of a high school diploma. We are skeptical that that is the still the case. We would like to test (using our sample from #7), whether there is evidence to suggest that the the proportion of counties with a median education level of high school diploma is diffrent from 45%.

### 9.a.  [1 pt] Formulate your null and alternative hypothses for this hypothesis test.

h0: \$p\$ = .45

ha: \$p\$ != .45

### 9b.  [1 pt] Check the conditions for conducting hypothesis testing using the sample from #7.



Sample is collected randomly, n is less than 10% of population size, np: (.45 * 40) > 10, and n*(1-p): (40*.55) >= 10

### 9c.  [1 pt] Calculate the p-value for this test.

Assume you do not know anything about the population proportion (including the value of $p$).

In [45]:
n_prop = 40

assumed_pop_prop1 = .45
print('Assumed Population Proportion:', assumed_pop_prop1)

std_error_prop1 = np.sqrt((assumed_pop_prop1 * (1 - assumed_pop_prop1))/n_prop)
print('Standard Error: ', std_error_prop1)


Assumed Population Proportion: 0.45
Standard Error:  0.07866066361276136


In [46]:
p_value1 = 2 * (1 - norm.cdf(prop_sample_hs, loc = assumed_pop_prop1, scale = std_error_prop1))
print('p-value: ', p_value1)

p-value:  0.5250099664186034


the p-value is 0.5250099664186034

### 9d.  [1 pt] Make a conclusion about your hypotheses using this p-value and a significance level of $\alpha=0.10$.

Because the p-value = 0.5250099664186034 > $\alpha\$ = 0.10, we fail to reject the null hypothesis -- there is insufficient evidence to suggest the alternative hypothesis

### 9e.  [1 pt] Now we would like to make a 90% confidence interval for the proportion of ALL U.S. counties that have a median education level of high school diploma.. First, check the conditions for creating this confidence interval.

Sample is collected randomly, n is less than 10% of population size, np: (.45 * 40) > 10, and n*(1-p): (40*.55) >= 10

### 9f.  [2 pt] Now create this 90% confidence interval.

Assume you do not know anything about the population (including $p$).

In [47]:
alpha = 0.10
critical_value1 = norm.ppf(.95)
#equivalent to norm.ppf(.95)

print('Critical Value: ', critical_value1)





ll1 = prop_sample_hs - critical_value1 * np.sqrt(assumed_pop_prop1 * (1 - assumed_pop_prop1)/n_prop)
ul1 = prop_sample_hs + critical_value1 * np.sqrt(assumed_pop_prop1 * (1 - assumed_pop_prop1)/n_prop)

    #n_prop = 40
    #assumed_pop_prop1 = .45


print('(',ll1,',',ul1,')')
print(assumed_pop_prop1)



Critical Value:  1.6448536269514722
( 0.3706147221581398 , 0.6293852778418603 )
0.45


### 9g.  [1 pt] Use this confidence interval to make a conclusion about your hypothesis tests.

Because the null value \$p\$ = .45 is inside the confidence interval, we fail to reject the null -- there is insufficient evidence to suggest the alternative hypothesis